src/collision.cpp

```diff
@@ -280,9 +280,17 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 		MapNode n = map->getNodeNoEx(p, &is_position_valid);
 
 		if (is_position_valid) {
```

Comment:
Why not just make this ``if (is_position_valid && n.getContent() != CONTENT_IGNORE) {``?

Reply:
I considered that, but then 'any position valid' is not set to true for ignore despite ignore being a valid position, this would affect the code below:
```
	// Do not move if world has not loaded yet, since custom node boxes
	// are not available for collision detection.
	if (!any_position_valid) {
		*speed_f = v3f(0, 0, 0);
		return result;
	}
```

Reply:
Not sure if that matters or not though, but it might.

Reply:
I think it's fine to hit it in the case of only ignore being found. As far as movement goes, ignore really is essentially unloaded world. Doing it this way also enables the optimization of skipping the actual collision algorithm in case the object clearly has no business moving anywhere.

Just add `// This also intentionally occurs in the case of the object being positioned solely on CONTENT_IGNORE nodes, no matter where they come from.` before `if (!any_position_valid) {` to make this clear.

Reply:
Ok will do.

Reply:
Done, will retest.